### PR TITLE
[Merged by Bors] - Compiler support onnx to enflame

### PR DESCRIPTION
**Pull request by shimei0203**

### CODE REVIEW

File: model_compiler/src/model_compiler/models/targets/enflame_model.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            for model_output in self.model_outputs
+        ]
+
+    def save(self, path: str) -> None:
+        os.makedirs(path, exist_ok=True)
+        self.engine.save_executable(os.path.join(path, 'model.exec'))
+
+    @staticmethod
+    def get_platform() -> Tuple[str, str]:
+        return 'tops-inference', '2.0.96'
```
Comment (NaNAGISaSA) [inline model_compiler/src/model_compiler/models/targets/enflame_model.py:52]: is there an api to acquire the platform version?

File: model_compiler/src/model_compiler/compilers/__init__.py [hunk 2]
```diff
@@ -15,9 +15,10 @@
     openvino_model_file_to_openvino_model
 
 try:
```
Comment (NaNAGISaSA) [inline model_compiler/src/model_compiler/compilers/__init__.py:17]: You should split these try cases in to two `try ... except` code piece, think about this case: `onnx_model_to_tensorrt_model` causes an error, but `onnx_model_file_to_enflame_model` is ok.

File: model_compiler/src/model_compiler/models/targets/__init__.py [hunk 3]
```diff
@@ -6,9 +6,10 @@
 from . import saved_model, tflite_model, paddle_model
 
 try:
```
Comment (NaNAGISaSA) [inline model_compiler/src/model_compiler/models/targets/__init__.py:8]: same problem as above

File: model_compiler/tests/model_compiler/compilers/test_onnx_model_file_to_enflame_model.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+from unittest import TestCase
+
+import tensorflow as tf
+import onnx
+import pytest
+
+import model_compiler.compilers.tf_model_to_tf_frozen_graph_model as tf_model_compiler
+import model_compiler.compilers.tf_frozen_graph_model_to_onnx_model as frozen_graph_compiler
+import model_compiler.compilers.onnx_model_file_to_enflame_model as onnx_file_to_enflame_compiler
+# pylint: disable=no-name-in-module
```
Comment (NaNAGISaSA) [inline model_compiler/tests/model_compiler/compilers/test_onnx_model_file_to_enflame_model.py:14]: do not use global lint disable

File: model_compiler/tests/model_compiler/compilers/test_onnx_model_file_to_enflame_model.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+from model_compiler.compilers.onnx_model_file_to_enflame_model import Config, DataFormat
+# pylint: disable=no-name-in-module
```
Comment (NaNAGISaSA) [inline model_compiler/tests/model_compiler/compilers/test_onnx_model_file_to_enflame_model.py:17]: do not use global lint disable

File: model_compiler/tests/model_compiler/models/targets/test_enflame_model.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+from unittest import TestCase
+
+import tensorflow as tf
+import onnx
+import onnx.utils
+import pytest
+import model_compiler.compilers.tf_model_to_tf_frozen_graph_model as tf_model_compiler
+import model_compiler.compilers.tf_frozen_graph_model_to_onnx_model as frozen_graph_compiler
+import model_compiler.compilers.onnx_model_file_to_enflame_model as enflame_model_compiler
+# pylint: disable=no-name-in-module
```
Comment (NaNAGISaSA) [inline model_compiler/tests/model_compiler/models/targets/test_enflame_model.py:16]: do not use global lint disable

File: model_compiler/tox.ini [hunk 7]
```diff
@@ -10,7 +10,7 @@ commands = {envpython} tools/install-tensorrt.py
            mypy --install-types --non-interactive -p model_compiler -p tests.model_compiler
            pylint src tests
 
-           cpu: pytest -m 'not gpu_test' --cov --cov-config .coveragerc-cpu -n auto
+           cpu: pytest -m 'not gpu_test and not dtu_test' --cov --cov-config .coveragerc-cpu -n auto
```
Comment (NaNAGISaSA) [inline model_compiler/tox.ini:13]: Maybe we should add another test suit named dtu?

**Review by NaNAGISaSA — CHANGES_REQUESTED** [review]

File: model_compiler/tox.ini (hunk 7 above)
Comment (shimei0203) [reply model_compiler/tox.ini:13]: There is no good solution here for the time being, other places have been changed. thank you

Comment (NaNAGISaSA) [reply model_compiler/tox.ini:13]: @shimei0203  I mean that we should add another test suit as cpu and nvidia gpu, for example: 
```
cpu: pytest ...
gpu: pytest ...
dtu: pytest ...
```
The reason for this is that gpu test suits can not run on dtu hardware, so I think we should distinguish the tests between gpu and dtu.

File: model_compiler/tests/model_compiler/models/targets/test_enflame_model.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+        self.assertEqual(engine.get_outputs(), [ModelOutput(name='z:0', data_type='DT_FLOAT', dims=[4])])
+
+    def test_save(self):
+        engine = _make_enflame_model()
+
+        with TemporaryDirectory() as save_path:
+            engine.save(save_path)
+
+            self.assertEqual(os.listdir(save_path), ['model.exec'])
```
Comment (NaNAGISaSA) [inline model_compiler/tests/model_compiler/models/targets/test_enflame_model.py:69]: Better to place this assert outside `with` scope.

**Review by NaNAGISaSA — CHANGES_REQUESTED** [review]

**Review by NaNAGISaSA — APPROVED** [review]

**Pull request closed without merging**